models/core/src/migration.ts

```diff
@@ -258,6 +253,149 @@ async function processMigrateContentFor (
   }
 }
 
+async function migrateCollaborativeDocsToJson (client: MigrationClient): Promise<void> {
+  const ctx = new MeasureMetricsContext('migrateCollaborativeDocsToJson', {})
+  const storageAdapter = client.storageAdapter
+
+  const hierarchy = client.hierarchy
+  const classes = hierarchy.getDescendants(core.class.Doc)
+  for (const _class of classes) {
+    const domain = hierarchy.findDomain(_class)
+    if (domain === undefined) continue
+
+    const allAttributes = hierarchy.getAllAttributes(_class)
+    const attributes = Array.from(allAttributes.values()).filter((attribute) => {
+      return hierarchy.isDerived(attribute.type._class, core.class.TypeCollaborativeDoc)
+    })
+
+    if (attributes.length === 0) continue
+    if (hierarchy.isMixin(_class) && attributes.every((p) => p.attributeOf !== _class)) continue
+
+    const query = hierarchy.isMixin(_class) ? { [_class]: { $exists: true } } : { _class }
+
+    const iterator = await client.traverse(domain, query)
+    try {
+      console.log('processing', _class)
```

Comment:
Minor: use `ctx.info` instead if you wanna keep these logs? There are more console.logs across the file.

Reply:
Done